volmaster/daemon.go

```diff
@@ -129,16 +134,32 @@ func (d daemonConfig) handleUnmount(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
-func (d daemonConfig) handleMount(w http.ResponseWriter, r *http.Request) {
+func (d daemonConfig) handleMountWithTTLFlag(w http.ResponseWriter, r *http.Request, exist client.PrevExistType) error {
 	req, err := unmarshalUseConfig(r)
 	if err != nil {
-		httpError(w, "Unmarshalling request", err)
-		return
+		return err
 	}
 
 	req.Reason = "Mount"
 
-	if err := d.config.PublishUse(req); err != nil {
+	fmt.Println(req)
+
+	if err := d.config.PublishUseWithTTL(req, time.Duration(d.mountTTL)*time.Second, exist); err != nil {
```

Comment:
I agree, but volplugin handling the updates is less desirable; this allows volmasters to work directly against etcd without sharing credentials that the volmaster/volplugin may share, ensuring a security model with cert auth and so forth.

I would argue we leave this for now, and introduce some kind of general configuration for what will eventually be called "Contiv Storage". This configuration could live in etcd and be passed through the REST interface so that volcli and volplugin can interact with the volmaster to get and change this information.

I can put in an issue for this now if it seems to you like a reasonable replacement. The TTL (and other settings, like debug) could be stored there so it could be referenced by both services.